plugins/techdocs-addons/src/context.tsx

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CompoundEntityRef } from '@backstage/catalog-model';
+import { useApi, useApp } from '@backstage/core-plugin-api';
+import {
+  techdocsApiRef,
+  TechDocsEntityMetadata,
+  TechDocsMetadata,
+} from '@backstage/plugin-techdocs';
+import React, {
+  createContext,
+  Dispatch,
+  PropsWithChildren,
+  SetStateAction,
+  useContext,
+  useState,
+} from 'react';
+import useAsync from 'react-use/lib/useAsync';
+
+type PropsWithEntityName = PropsWithChildren<{ entityName: CompoundEntityRef }>;
+
+const TechDocsMetadataContext = createContext<TechDocsMetadata | undefined>(
+  undefined,
+);
+
+export const TechDocsMetadataProvider = ({
+  entityName,
+  children,
+}: PropsWithEntityName) => {
+  const { NotFoundErrorPage } = useApp().getComponents();
+  const techdocsApi = useApi(techdocsApiRef);
+
+  const { value, loading, error } = useAsync(async () => {
+    return await techdocsApi.getTechDocsMetadata(entityName);
+  }, []);
+
+  if (!loading && error) {
```

Comment:
what do you think about making the loading and error states available through the context instead and let the ones consuming it be responsible for error handling?